src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -52,6 +52,13 @@ Qgs3DMapCanvas::Qgs3DMapCanvas( QWidget *parent )
     emit savedAsImage( mCaptureFileName );
   } );
 
+  connect( mEngine, &QgsAbstract3DEngine::depthBufferCaptured, [ = ]( const QImage & image )
+  {
+    mDepthBufferImage = image;
```

Comment:
Is it necessary that Qgs3DMapCanvas keeps a copy of the depth buffer image? I would suggest to remove `mDepthBufferImage` if not...

Reply:
Initially I wanted to have the depth buffer copied every N frame but that doesn't work well with the on demand rendering and unnecessary..